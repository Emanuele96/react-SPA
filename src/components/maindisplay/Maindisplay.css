
.maindisplay {
  border: rgb(0, 255, 255) dashed 1px;
  height: 100%;
  width: 100%;
  padding: 1rem 0;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
}

.favorite-buttons {
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.btn {
  height: 50px;
  width: auto;
  padding: 1rem;
  border-radius: 25px;
  text-decoration: none;
  cursor: pointer;
}

.btn:focus {
  outline: none;
}

.btn h3 {
  margin: 0;
  padding: 0;
}

.favorite {
  background-color: rgb(255, 0, 0, 0.4);
  transition: background-color ease-in-out 150ms;
}

.favorite:hover {
  background-color: rgb(255, 0, 0, 0.65);
  transition: background-color ease-in-out 150ms;
}

.delete {
  background: none;
  border: solid 2px rgb(255, 0, 0, 0.3);
  transition: border ease-in-out 150ms;
}

.delete:hover {
  border: solid 2px rgb(255, 0, 0, 0.7);
  transition: border ease-in-out 150ms;
}
