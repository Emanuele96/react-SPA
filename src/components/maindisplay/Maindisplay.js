import React from 'react';
import './Maindisplay.css';
import AudioPlayer from '../soundform/AudioPlayer';

const Maindisplay = props => {
  return (
    <div className='Maindisplay'>
      <WelcomeMessage
        selectedTab={props.selectedTab}
        soundCategory={props.soundCategory}
        soundTrack={props.soundTrack}
      />
      <div className='favorite-buttons'>
        <button className='btn delete'>
          <h3>Slett favoritt kombinasjon</h3>
        </button>
        <button className='btn favorite'>
          <h3>Lagre Kombinasjon</h3>
        </button>
      </div>
    </div>
  );
};

const WelcomeMessage = props => {
  return (
    <div>
      <h2>
        Velkommen! {props.selectedTab && `Kombinasjon: ${props.selectedTab}`}
      </h2>
      <p>Vennligst velg tre kategorier per medietype i vindu til h√∏yre.</p>
      <AudioPlayer
        soundTrack={props.selectedTab}
        soundCategory={props.soundCategory}
      />
    </div>
  );
};

export default Maindisplay;
