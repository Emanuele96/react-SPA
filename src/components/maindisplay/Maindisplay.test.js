import React from 'react';
import renderer from 'react-test-renderer';

import Maindisplay from './Maindisplay';



it('renders correctly ', () => {
    const tree = renderer.create(<Maindisplay/>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly when passing multiple props', () =>{
    const tree = renderer.create(<Maindisplay
         selectedTab="1" 
         soundCategory="Nature"
         />).toJSON();
    expect(tree).toMatchSnapshot();
});

