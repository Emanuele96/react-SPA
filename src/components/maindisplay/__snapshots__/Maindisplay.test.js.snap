// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly  1`] = `
<div
  className="maindisplay"
>
  <div>
    <h2>
      Velkommen! 
    </h2>
    <p>
      Vennligst velg tre kategorier per medietype i vindu til høyre.
    </p>
    <div
      className="sound"
    >
      <audio
        autoPlay={true}
        className="soundPlayer"
        controls={true}
        loop={true}
        preload="auto"
        type="audio/mpeg"
      />
    </div>
  </div>
  <div
    className="favorite-buttons"
  >
    <button
      className="btn delete"
    >
      <h3>
        Hent favoritt kombinasjon
      </h3>
    </button>
    <button
      className="btn favorite"
    >
      <h3>
        Lagre Kombinasjon
      </h3>
    </button>
  </div>
</div>
`;

exports[`renders correctly when passing multiple props 1`] = `
<div
  className="maindisplay"
>
  <div>
    <h2>
      Velkommen! 
      Kombinasjon: 1
    </h2>
    <p>
      Vennligst velg tre kategorier per medietype i vindu til høyre.
    </p>
    <div
      className="sound"
    >
      <audio
        autoPlay={true}
        className="soundPlayer"
        controls={true}
        loop={true}
        preload="auto"
        src="1.mp3"
        type="audio/mpeg"
      />
    </div>
  </div>
  <div
    className="favorite-buttons"
  >
    <button
      className="btn delete"
    >
      <h3>
        Hent favoritt kombinasjon
      </h3>
    </button>
    <button
      className="btn favorite"
    >
      <h3>
        Lagre Kombinasjon
      </h3>
    </button>
  </div>
</div>
`;
